src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java

```diff
@@ -1245,4 +1249,260 @@ public SBApiResponse workflowBulkUpdateTransitionV1(String userAuthToken, Multip
 		}
 		return response;
 	}
+
+	@Override
+	public ResponseEntity<?> downloadPendingRequestFile(String userAuthToken) {
+		ResponseEntity<?> responseEntity;
+		try {
+			Set<String> userIdSet = new HashSet<>();
+			Workbook pendingRequestWorkBook = this.getBlankWorkseet();
+			String mdoUserId = accessTokenValidator.fetchUserIdFromAccessToken(userAuthToken);
+			userIdSet.add(mdoUserId);
+			Map<String, Object> mdoDetails = this.getUserSearchDetails(userIdSet,true, null);
+			Map<String, Object> mdoInfo = (Map<String, Object>) mdoDetails.get(mdoUserId);
+			String departmentName = (String) mdoInfo.get(Constants.ROOT_ORG_NAME);
+			String rootOrgId = (String) mdoInfo.get(Constants.ROOT_ORG_ID);
+			Map<String, Object> allPendingRequest = new HashMap<>();
+			List<WfStatusEntity> pendingRequestsEntityList = wfStatusRepo.getListOfDistinctApplicationUsingDept(Constants.PROFILE_SERVICE_NAME, Constants.SEND_FOR_APPROVAL, departmentName);
+			if(CollectionUtils.isEmpty(pendingRequestsEntityList)) {
+				return ResponseEntity.status(HttpStatus.OK).body("There are no approval requests pending for approval with the MDO");
```

Comment:
Let's return proper response body instead of returning text string... 
if there are any pending request -- we return list of objects... in case of no pending approvals -- return empty list.

Reply:
I am returning a csv file with proper message in case there are no pending requests.